prov/usnic/src/usdf_ep_rdm.c

```diff
@@ -878,16 +878,29 @@ static int
 usdf_rx_rdm_port_bind(struct usdf_rx *rx, struct fi_info *info)
 {
 	struct sockaddr_in *sin;
+	struct sockaddr_in *tmp_sin;
 	struct sockaddr_in src;
 	socklen_t addrlen;
 	int ret;
+	bool addr_format_str = false;
+	size_t addr_str_len = USDF_ADDR_STR_LEN;
 
 	if (info->src_addr != NULL) {
-		if (info->addr_format != FI_SOCKADDR &&
-		    info->addr_format != FI_SOCKADDR_IN) {
+		switch (info->addr_format) {
```

Comment:
This block looks similar to an earlier one, couldn't it be rolled into a function?